core/src/types/operator/operator.rs

```diff
@@ -1254,6 +1254,25 @@ impl Operator {
     /// # }
     /// ```
     ///
+    /// ## `if_not_exist`
+    ///
+    /// Set `if_not_exist` for this `write` request. This can be treated as a simplified version
+    /// of [`OpWrite::if_none_match`].
+    ///
+    /// This feature is used to write a file only when it doesn't exist. If the file already
+    /// exists, an error with kind [`ErrorKind::ConditionNotMatch`] will be returned.
+    ///
```

Comment:
We can clerify that `if_not_exist` is concurrency-safe and can be used similarly to a file lock. Storage services that support `if_not_exist` ensure that only one write succeeds while all other concurrent writes will fail.